### ISSUE TITLE
[CI] Add Docker status check

### ISSUE THREAD

**Issue by jasonLaster** [devtools-html/debugger.html#1255]
Our integration tests have been monotonically increasing in [run time](https://circleci.com/build-insights/gh/devtools-html/debugger.html/master). Our median run time is now 15 minutes.

The lowest hanging fruit would be to improve the docker setup time. The time comes from two steps

1) load from the docker VM cache (3 minutes)
2) download an update from the docker registry (6 minutes)

The update takes about 6 minutes, but can run over 10 minutes and fail on occasion, which is worse. We can fix this by first checking docker for the tag and comparing that against the docker image on our VM (~ 20 seconds).

This fix would likely bring our runtimes from ~15 minutes to ~9 minutes. And then we can figure out other wins after that.

**what's the best way to get started**

1. go to a previous circle [run](https://circleci.com/gh/devtools-html/debugger.html/tree/master)
2. re-run it with ssh rebuild
3. ssh into the machine
4. mess with docker commands and the update-docker script until it works

 🏃  here's a [gif](http://g.recordit.co/ylTMpT9EgB.gif) of me starting to do that

**Here's a patch with a sketch of the solution**

```diff
diff --git a/bin/update-docker b/bin/update-docker
index 7d7fe54..0f168d3 100755
--- a/bin/update-docker
+++ b/bin/update-docker
@@ -10,7 +10,12 @@ fi

 docker images

-time docker pull jlongster/mochitest-runner
+# compare the output of docker images
+# with the output of some docker query like
+# docker LS down here
+if [[ ]]; then
+ time docker pull jlongster/mochitest-runner
+fi

 docker images
```

cc @clarkbw , @juliandescottes

**Comment by jasonLaster** [comment]
One thought, while i'm thinking about it.

At this point, we don't update the docker image that often. We could change the script pretty easily so the docker pull only happens when the cache is empty...

Here's how we clear the [cache](https://cloud.githubusercontent.com/assets/254562/20451908/38fbf28e-adcd-11e6-9032-bd07e9a79bd5.png)

Here's a sketch of that idea

```diff
diff --git a/bin/update-docker b/bin/update-docker
index 7d7fe54..39d6096 100755
--- a/bin/update-docker
+++ b/bin/update-docker
@@ -5,12 +5,11 @@ DOWNLOADS_PATH=${DOWNLOADS_PATH:-"$HOME/downloads"}
 mkdir -p $DOWNLOADS_PATH

 if [[ -e $DOWNLOADS_PATH/mochitest-runner.tar ]]; then
-  time docker load -i $DOWNLOADS_PATH/mochitest-runner.tar;
+    time docker pull jlongster/mochitest-runner
 fi

 docker images

-time docker pull jlongster/mochitest-runner

 docker images

```

**Comment by jasonLaster** [comment]
I took another look with `ssh-build` and found two things:

1. if i tag the image id, the image is 752.8mb. 

11.25 GB seems really large, which makes me think that perhaps it saves each of the previous versions? If we could bring the size down to 4.3 GB, which is what my `obj-frontend` is then we'd be in a better spot. Also, i [scanned](https://cloud.githubusercontent.com/assets/254562/20452798/2f7056ca-ade0-11e6-9f3c-de1109d9b6da.png)
 the dir to see if there was cruft we could remove and didn't find any likely candidates.


```
docker tag 4390fc091451 mochitest-runner/11.18.16
```

```
REPOSITORY                   TAG                 IMAGE ID            CREATED             VIRTUAL SIZE
jlongster/mochitest-runner   latest              4390fc091451        3 weeks ago         11.25 GB
mochitest-runner/11.18.16    latest              a93b25789168        4 weeks ago         752.8 MB
```

2. I took a look at the times for each operation in the script 

I was surprised to see that the pull time was negligent. The real time is load and save. If we know that the pull was a no-op, we can avoid saving, which is like the same thing as before but with saving be the step we avoid.

* *docker load*  3 minutes
* *docker pull*  30 seconds
* *docker save* 6 minutes

Ideally we can do two things:

1. cut the image size down significantly
2. avoid saving unchanged images
